test_html3.py

```diff
@@ -0,0 +1,86 @@
+"""Tests for html server, v3. Starts server with fixture."""
+import os
+import sys
+import pytest
+import urllib2
+from html_server import html_response
+import threading
+
+
+def test_server_home(start_server):
+    """Test that html server returns directory listing as html."""
+    response = urllib2.urlopen('http://127.0.0.1:50000/')
+    assert response.read() == '<!DOCTYPE html><html><ul><p>. contains:</p><li><a href="./a_web_page.html">a_web_page.html</a></li><li><a href="./images">images</a></li><li><a href="./make_time.py">make_time.py</a></li><li><a href="./sample.txt">sample.txt</a></li></ul></html>'
+
+
+def test_server_html(start_server):
+    """Test that html server returns HTML file."""
+    response = urllib2.urlopen('http://127.0.0.1:50000/a_web_page.html').read()
+    assert response == '<!DOCTYPE html>\n<html>\n<body>\n\n<h1>North Carolina</h1>\n\n<p>A fine place to spend a week learning web programming!</p>\n\n</body>\n</html>\n\n'
+
+
+def test_server_txt(start_server):
+    """Test that html server returns txt file."""
+    response = urllib2.urlopen('http://127.0.0.1:50000/sample.txt').read()
+    assert response == 'This is a very simple text file.\nJust to show that we can server it up.\nIt is three lines long.\n'
+
+
+def test_server_file(start_server):
+    """Test that html server returns JPEG."""
+    response = urllib2.urlopen('http://127.0.0.1:50000/images/JPEG_example.jpg').read()
+    os.chdir('/Users/mark/projects/network_tools/webroot')
+    actual = open('images/JPEG_example.jpg', 'rb').read()
+    assert actual == response
```

Comment:
This test shouldn't work, as the response should have http headers in it that are not in the image at all.  Does it work?